experimental/dds/tree/src/EditLog.ts

```diff
@@ -425,27 +469,24 @@ export class EditLog<TChange = unknown> extends TypedEventEmitter<IEditLogEvents
 		info?: EditSequencingInfo,
 		minSequenceNumber: number = 0
 	): void {
-		assert(
-			minSequenceNumber >= this.minSequenceNumber,
-			'Sequenced edits should carry a monotonically increasing min number'
-		);
-		// The new minSequenceNumber indicates that no future edit will require information from edits with a smaller or equal seq number
-		// for its resolution.
-		this._minSequenceNumber = minSequenceNumber;
-		// TODO:#57176: Increment maximumEvictableIndex to reflect the fact we can now evict edits with a sequenceNumber lower or equal to
-		// it. Note that this will change the meaning of our 'InSession' APIs so we should make sure to rename them at the same time.
-		// The code might look like this:
-		// while (this.maximumEvictableIndex + 1 < this.indexOfFirstEditInSession) {
-		// 	const nextEdit = this.getEditInSessionAtIndex(this.maximumEvictableIndex + 1);
-		// 	const nextEditInfo = this.getOrderedEditId(nextEdit.id) as SequencedOrderedEditId;
-		// 	if (
-		// 		nextEditInfo.sequenceInfo !== undefined &&
-		// 		nextEditInfo.sequenceInfo.sequenceNumber > minSequenceNumber
-		// 	) {
-		// 		break;
-		// 	}
-		// 	++this.maximumEvictableIndex;
-		// }
+		// The index of the sequenced edit to add
+		const index = this._earliestAvailableEditIndex + this.numberOfSequencedEdits;
+
+		if (info !== undefined) {
+			assert(
+				minSequenceNumber >= this._minSequenceNumber,
+				'Sequenced edits should carry a monotonically increasing min number'
+			);
+			this._minSequenceNumber = minSequenceNumber;
+
+			// Calculate the index of the minSequenceNumber by comparing it to the index of this sequenced edit
+			const sequenceNumberDifference = info.sequenceNumber - minSequenceNumber;
+			const minSequenceIndex = sequenceNumberDifference > 0 ? index - sequenceNumberDifference : index;
+
+			// The new minSequenceIndex indicates that no future edit will require information from edits with a smaller or equal index
+			// for its resolution.
+			this._minSequenceIndex = minSequenceIndex > 0 ? minSequenceIndex : 0;
```

Comment:
nit: `Math.max(0, minSequenceIndex)`

If you strongly prefer ternary over Math.min/max, I'm fine with you keeping it, but Math.min/max is terser and usually more readable imo

Reply:
I totally did not realize it works here lol. Will definitely change.